lib/tasks/bundlers/utils/applyDefaultsToBundleDefinition.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Applies default values to bundleDefinitions
+ *
+ * @param {module:@ui5/builder/processors/bundlers/moduleBundler~ModuleBundleDefinition} bundleDefinition Module
+ *		bundle definition
+ * @param {@ui5/project/build/helpers/TaskUtil|object} [taskUtil] TaskUtil
+ *
+ * @returns {module:@ui5/builder/processors/bundlers/moduleBundler~ModuleBundleDefinition}
+ */
+export function applyDefaultsToBundleDefinition(bundleDefinition, taskUtil) {
+	bundleDefinition.sections = bundleDefinition?.sections?.map((section) => {
+		const defaultValues = {
+			resolve: false,
+			resolveConditional: false,
+			renderer: false,
+			sort: true,
+			declareRawModules: false,
+		};
+
+		// Since specVersion: 4.0 "require" section starts loading asynchronously.
+		// If specVersion cannot be determined, the latest spec is taken into account.
+		// This is a breaking change in specVersion: 4.0
+		if (section.mode === "require" && (!taskUtil || taskUtil.getProject().getSpecVersion().gte("4.0"))) {
+			defaultValues.async = true;
```

Comment:
Is there a test that verifies this behavior?
The bundle tooling already defaults to "async=true", so I doubt that this line has any effect. My expectation would be that the flag has to be actively set to false in the else case, as this is the value that differs from the new default behavior.

I tested this with a self-contained build and even with specVersion 3.0 I get an async require section, which is not expected.

Reply:
We should also document the "async" option in the respective JSDoc: https://github.com/SAP/ui5-builder/blob/main/lib/processors/bundlers/moduleBundler.js#L13